### ISSUE TITLE
Build command hardcodes config/app.php values

### ISSUE THREAD

**Issue by CupOfTea696** [laravel-zero/laravel-zero#332]
The build command hardcodes the computed values from `config/app.php` in Phar builds, which may interfere with app functionality.

Example `config/app.php`:
```php
    /*
    |--------------------------------------------------------------------------
    | Encryption Key
    |--------------------------------------------------------------------------
    |
    | This key is used by the Illuminate encrypter service and should be set
    | to a random, 32 character string, otherwise these encrypted strings
    | will not be safe. Please do this before deploying an application!
    |
    */

    'key' => env('APP_KEY'),

    'cipher' => 'AES-256-CBC',
```

Would be `phar://…/builds/app/config/app.php`:
```php
    /*
    |--------------------------------------------------------------------------
    | Encryption Key
    |--------------------------------------------------------------------------
    |
    | This key is used by the Illuminate encrypter service and should be set
    | to a random, 32 character string, otherwise these encrypted strings
    | will not be safe. Please do this before deploying an application!
    |
    */

    'key' => 'MY_APP_KEY',

    'cipher' => 'AES-256-CBC',
```

This could potentially be solved by using regex to replace or set the 'version' key in the config, instead of using `include` and `var_export`.

**Comment by owenvoke** [comment]
Hmm. I've definitely tried to think of a good way to implent this before.

I've only used the `app.key` property once before in a Laravel Zero app (as it's only used when `illuminate/encryption` is used), and in that I don't compile to a Phar. For nearly all custom settings, I try to keep them in another configuration file to avoid this issue. `var_export()` is a pretty clean way of doing this, however I agree that there's probably a better way of doing this to preserve the configuration. 🤔

Using RegEx may be quite awkward, due to the `env` and `version` being set. We could obviously replace `/'version' => app\('git\.version'\),/` and `'env' => 'development',` with the relevant values using RegEx. However, for example, I have a custom config with `'env' => env('V_APP_ENV', 'development'),` and other people could have countless custom versions. The current method of changing the array values and exporting definitely is the easiest way to cover all options.

---

Also, on a side note @nunomaduro, is [this line](https://github.com/laravel-zero/framework/blob/master/src/Commands/BuildCommand.php#L173) ad duplicate of [this one](https://github.com/laravel-zero/framework/blob/master/src/Commands/BuildCommand.php#L165), or are they both needed?

**Comment by CupOfTea696** [comment]
@owenvoke Yeah, I've been looking into this and using an actual PHP parser would be a much safer option, but is obviously a lot more complex than `var_export()`.

**Comment by ericlucit** [comment]
We get around this issue by calling `$mySecretSetting = env('MY_SETTING_NAME');` from outside of the config files directly in the code that requires it.  This prevents the settings from being built compiled into the app.

We can then store these variables in `.env` files located in the same directory as the phar archive

We also use a preflight check when the application launches to ensure that the `.env` file exists and it contains the expected setttings.

in a nutshell, instead of `$setting = config("app.setting");` we use `$setting  = env('THE_SETTING');`

**Comment by owenvoke** [comment]
Interesting. I think this only applies to `app.*` settings, however I do the same thing (except I just put them in another config file).

For example, with the [Oh Dear CLI](https://github.com/nunomaduro/ohdear-cli/blob/d2da299b0e511c9e0425f0d09037fa99f71edbb1/config/settings.php#L17) the custom configuration is in a `settings.*` file.

Then you can carry on using `config('settings.*')` wherever you need to rather than duplicating it in the code. 👍🏻

**Comment by ericlucit** [comment]
> Interesting. I think this only applies to `app.*` settings, however I do the same thing (except I just put them in another config file).
> Then you can carry on using `config('settings.*')` wherever you need to rather than duplicating it in the code. 👍🏻

Ok - This is good to know! - So, to confirm.  Creating a new config, e.g. `myapp.php` in the config folder, the builder will NOT compile these settings into the phar if they come from `env(...)`

**Comment by owenvoke** [comment]
Yeah, it's only for the `config/app.php`
